src/GitVersion.Core.Tests/IntegrationTests/PreventDecrementationOfVersionsOnTheDevelopBranchScenario.cs

```diff
@@ -0,0 +1,165 @@
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configurationBuilder.Build());
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configurationBuilder.Build());
+
+        // now we do the actual release of beta 1
+        fixture.Checkout("release/1.0.0");
+        fixture.ApplyTag("1.0.0-beta.1");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1", configurationBuilder.Build());
+
+        // continue with more work on develop that may or may not end up in the 1.0.0 release
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configurationBuilder.Build());
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.2", configurationBuilder.Build());
+
+        // now we decide that the new on develop should be part of the beta 2 release
+        // se we merge it into release/1.0.0 with --no-ff because it is a protected branch
+        // but we don't do the release of beta 2 just yet
+        fixture.Checkout("release/1.0.0");
+        fixture.MergeNoFF("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2+2", configurationBuilder.Build());
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configurationBuilder.Build());
+
+        fixture.Checkout("release/1.0.0");
+        fixture.ApplyTag("1.0.0-beta.2");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configurationBuilder.Build());
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configurationBuilder.Build());
+
+        fixture.MergeNoFF("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.3", configurationBuilder.Build());
+
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.3", configurationBuilder.Build());
+
+        configurationBuilder.WithNextVersion("1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.3", configurationBuilder.Build());
+
+        fixture.Repository.Tags.Remove("1.0.0-beta.1");
+        fixture.Repository.Tags.Remove("1.0.0-beta.2");
+
+        // ❌ expected: "1.0.0-alpha.3"
+        // This behavior needs to be changed for the git flow workflow using the track-merge-message or track-merge-target options.
+        // [Bug] track-merge-changes produces unexpected result when combining hotfix and support branches #3052
+        fixture.AssertFullSemver("1.1.0-alpha.3", configurationBuilder.Build());
```

Comment:
I agree with you that it's not so easy to understand the use case at a glance. That's even true for all integration tests ;). I have reused the integration test from the user bitbonk in issue #3177 . I think if some one wants to understand this scenario the best way is to take a look into it.

Reply:
I agree some integration tests can become unwieldy, but I think most of our tests are easy to understand and follow. It would be awesome if @bitbonk could cooperate on creating a more condensed and terse integration test that clearly points out the problem in v5 that is now going to be fixed in v6 with this PR.

Reply:
Done